cmd/snap-mgmt/snap-mgmt.sh.in

```diff
@@ -26,20 +26,32 @@ SNAP_UNIT_PREFIX="$(systemd-escape -p ${SNAP_MOUNT_DIR})"
 
 systemctl_stop() {
     unit="$1"
-    if systemctl is-active -q "$unit"; then
-        echo "Stoping $unit"
+    for i in $(seq 10); do
+        if systemctl is-active -q "$unit"; then
+            echo "Stoping $unit"
+            systemctl stop -q "$unit" || true
+        fi
+        echo "Stoping $unit [attempt $i]"
         systemctl stop -q "$unit" || true
-    fi
+        sleep .1
+    done
 }
 
 purge() {
-    # undo any bind mount to ${SNAP_MOUNT_DIR} that resulted from LP:#1668659
-    if grep -q "${SNAP_MOUNT_DIR} ${SNAP_MOUNT_DIR}" /proc/self/mountinfo; then
-        umount -l "${SNAP_MOUNT_DIR}" || true
+    if grep -q CODENAME=trusty /etc/os-release >/dev/null 2>&1 ; then
+        # snap.mount.service is a trusty thing
```

Comment:
Please do not refer to distribution releases by specific codenames, refer to them by their full names. Context is important for a multi-distribution script.